src/main/java/ch/njol/skript/conditions/CondScriptLoaded.java

```diff
@@ -59,23 +59,19 @@
 	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
 		scripts = (Expression<String>) exprs[0];
 		setNegated(matchedPattern == 1);
-		assert getParser().getCurrentScript() != null;
-		if (getParser().isCurrentEvent(EffectCommandEvent.class) && scripts == null) {
-			Skript.error("The condition 'script loaded' requires a script name argument when used in effect commands");
+		Config cs = getParser().getCurrentScript();
+		if (cs == null && scripts == null) {
+			Skript.error("The condition 'script loaded' requires a script name argument when used outside of script files");
 			return false;
 		}
-		Config cs = getParser().getCurrentScript();
-		if (cs != null)
-			currentScriptFile = cs.getFile();
+		currentScriptFile = cs.getFile();
 		return true;
 	}
 	
 	@Override
 	public boolean check(Event e) {
 		Expression<String> scripts = this.scripts;
 		if (scripts == null) {
-			if (currentScriptFile == null)
-				return isNegated();
 			return ScriptLoader.getLoadedFiles().contains(currentScriptFile);
```

Comment:
IMO it's better not to check if it contains null, rather skip the check like you did before

Reply:
Sure \:)